kubernetesclient/authorization.go

```diff
@@ -0,0 +1,55 @@
+package kubernetesclient
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+const (
+	caLocation = "/etc/kubernetes/ssl/ca.pem"
+)
+
+var (
+	token  string
+	caData []byte
+)
+
+func init() {
+	if err := func() error {
```

Comment:
Can you just turn this into a normal `Init` function an call from main first?  It's just bad form to do I/O in an init, also this has a weird behavior that it will just block forever if you run the program not knowing stdin should have the token.